akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -280,42 +293,62 @@ private[akka] class RemoteActorRefProvider(
-    import remoteSettings._
-    val failureDetector = createRemoteWatcherFailureDetector(system)
+  /** Will call the provided `func` if using Cluster or explicitly enabled unsafe remote features. */
+  private def createOrNone[T](func: => T): Option[T] = if (hasClusterOrUseUnsafe) Some(func) else None
+
+  /**
+   * If using Cluster a `ClusterRemoteWatcher` is returned by the Cluster implementation.
+   * Or if not using Cluster and `akka.remote.use-unsafe-remote-features-without-cluster`
+   * is `on`, a `RemoteWatcher` is returned.
+   * If not using Cluster and `akka.remote.use-unsafe-remote-features-without-cluster`
+   * is `off` (default) then this function is not called during initialization, and no
+   * watcher is created. Additionally in this last case, no remote watcher
+   * `FailureDetector` is created or needed.
+   */
+  protected def createRemoteWatcher(system: ActorSystemImpl): ActorRef =
     system.systemActorOf(
-      configureDispatcher(
-        RemoteWatcher.props(
-          failureDetector,
-          heartbeatInterval = WatchHeartBeatInterval,
-          unreachableReaperInterval = WatchUnreachableReaperInterval,
-          heartbeatExpectedResponseAfter = WatchHeartbeatExpectedResponseAfter)),
+      RemoteWatcher.props(remoteSettings, createRemoteWatcherFailureDetector(system)),
       "remote-watcher")
-  }
-
-  protected def createRemoteWatcherFailureDetector(system: ExtendedActorSystem): FailureDetectorRegistry[Address] = {
-    def createFailureDetector(): FailureDetector =
-      FailureDetectorLoader.load(
-        remoteSettings.WatchFailureDetectorImplementationClass,
-        remoteSettings.WatchFailureDetectorConfig,
-        system)
 
-    new DefaultFailureDetectorRegistry(() => createFailureDetector())
-  }
+  protected def createRemoteWatcherFailureDetector(system: ExtendedActorSystem): FailureDetectorRegistry[Address] =
+    new DefaultFailureDetectorRegistry(() => FailureDetectorLoader(system, remoteSettings))
 
   protected def createRemoteDeploymentWatcher(system: ActorSystemImpl): ActorRef =
-    system.systemActorOf(
-      remoteSettings.configureDispatcher(Props[RemoteDeploymentWatcher]()),
-      "remote-deployment-watcher")
+    system.systemActorOf(RemoteDeploymentWatcher.props(remoteSettings), "remote-deployment-watcher")
 
-  /** Can be overridden when using RemoteActorRefProvider as a superclass rather than directly */
-  protected def showDirectUseWarningIfRequired() = {
+  /** Can be overridden when using RemoteActorRefProvider as a superclass rather than directly. */
+  protected def warnIfDirectUse(): Unit =
     if (remoteSettings.WarnAboutDirectUse) {
       log.warning(
         "Using the 'remote' ActorRefProvider directly, which is a low-level layer. " +
         "For most use cases, the 'cluster' abstraction on top of remoting is more suitable instead.")
     }
-  }
+
+  // Log on startup of provider.
+  def warnIfUseUnsafeWithoutCluster(): Unit =
```

Comment:
should `if (warnOnUnsafeRemote)` be used here too?

Reply:
I'd prefer to log it the one time on `init` as an initial warning, similar to `warnIfDirectUse`. It is the repetitive ones that the user should be able to dampen or move to using Cluster.

Reply:
ok, makes sense